### PR TITLE
Add ProtectedEntityID and BackupRepositoryName to Download CR

### PR DESCRIPTION
**Pull request by xing-yang**
Add ProtectedEntityID and BackupRepositoryName to Download CR.

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/93/

File: pkg/apis/veleroplugin/v1/download.go [hunk 1]
```diff
@@ -30,6 +30,13 @@ type DownloadSpec struct {
 	// The server's time is used for SnapshotTimestamp
 	// +optional
 	RestoreTimestamp *meta_v1.Time `json:"restoreTimestamp,omitempty"`
+
+	// ProtectedEntityID is the identifier for the protected entity.
```
Comment (dsu-igeek) [inline pkg/apis/veleroplugin/v1/download.go:34]: This is for the volume to overwrite, correct?  Would you add that to the comment?

Comment (xing-yang) [reply pkg/apis/veleroplugin/v1/download.go:33]: Sure.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/95/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/97/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/98/

**Pull request merged**